docs/3.0rc/resources/upgrade-prefect-3.mdx

```diff
@@ -0,0 +1,228 @@
+---
+title: Upgrade to Prefect 3
+description: Learn how to upgrade from Prefect 2 to Prefect 3.
+---
+
+Some patterns supported by Prefect 2 are no longer supported with Prefect 3, or are enabled differently. 
+This guide covers changes that you may need to make to ensure your Prefect 2 workflows keep flowing.
+
+<Info>
+Prefect 3 is an upgrade to the open source `prefect` Python package
+The Prefect 3 release does not involve any changes to Prefect Cloud.
+All previous and future Prefect 2 releases will continue to work with Prefect Cloud.
+</Info>
+
+## Module locations and names
+
+Some modules have been renamed, reorganized, or removed for clarity. 
+Objects imported from changed modules will continue to work until their six month deprecation period ends, but will raise deprecation warnings. 
+See the [migration script](https://github.com/PrefectHQ/prefect/blob/f767724ab28278446456b024b5c1c73f76ebcbb5/src/prefect/_internal/compatibility/migration.py#L22) for the full list of modules that have been moved and removed.
+
+## Asynchronous task execution behavior
+
+Prefect 2 abstracts away certain task execution concerns, which is convenient in some cases, but comes at the expense of unexpected behavior in other cases. 
```

Comment:
Another non-blocker - we may want to do a verb-tense sweep. Might help to refer to Prefect 2 in the past tense and Prefect 3 in the present/future.

Reply:
I'm reluctant to use past tense to refer to Prefect 2. It signals that Prefect 2 is already "in the past" but Prefect 3 this is just a release candidate. If I'm the minor here, I'll revise, but it was a conscious choice.